webrtc/RTCPeerConnection-transceivers.https.html

```diff
@@ -64,8 +64,8 @@
 
 promise_test(async t => {
   const pc = createPeerConnectionWithCleanup(t);
-  const track = await createTrackWithCleanup(t);
-  const sender = pc.addTrack(track);
+  const [track, stream] = await createTrackWithCleanup(t);
```

Comment:
If this returns both a track and a stream now, a name like createTrackAndStreamWithCleanup() makes more sense.

However, in the spirit of moving away from streams-with-tracks, you don't really need a stream where the track belongs to it. The relationship, as far as webrtc is concerned, is entirely described in terms of the senders and receivers.

What about not returning the stream, and instead doing this...?
```
pc.addTrack(track, new MediaStream());
```

That way we don't have to return an array of two different return values, and the helper method can do just one thing.

Assuming Firefox does not barf at that too. But this is optional, the "add[Track/Transceiver]([0/1/2] stream[/s])" tests already test the "new MediaStream()" semantics.

Reply:
last i checked the consensus was to avoid helper functions like this entirely. I won't clean up after you all the way :-p
Doing pc.addTrack(track, new MediaStream()) just to avoid limitations of a helper function seems even worse.

Reply:
renamed to createTrackAndStreamWithCleanup and squashed

Reply:
It's not to avoid a limitation of a helper function, it's to make the helper function do one thing: create a track that is cleaned up. The stream being created is an implementation detail we don't have to expose, and makes less sense when you want to have 0 or 2 streams for a track. It would be confusion to create multiple tracks just to get multiple streams, this is why we already do "new MediaStream()" in the "add[Track/Transceiver]([0/1/2] stream[/s])" tests, so I thought we could be consistent and do that everywhere.

Reply:
using new MediaStream() means adding a track that is not part of the stream. While that is probably useful to test it is somewhat artificial. ISTR that Firefox did not like this at some point but it seems to work in FF52 (ESR) even.

WRT to helpers it seems that there is also getNoiseStream which returns a stream. I'd prefer nott o increase the number of different semantics we use even more

Reply:
Whether the track is part of the stream or not is per-spec _completely irrelevant_. There is _absolutely nothing_ that says we should test this, other than "this is how the legacy stuff worked". It makes sense to have tests to make sure that the APIs work regardless of if the stream is empty, has the track, has another track, etc, just to make sure that common use cases are supported. But they should not be the basis for designing tests testing APIs that have nothing to say about what the contents of the stream is.

Therefor we should design our tests not with that in mind.

Reply:
So we should decide this based on: What makes more sense? A helper that gives us a track, or a helper that gives us both?

I argue the first makes slightly more sense, but I also realize I am bikeshedding. It's up to you.

Reply:
E_TOO_MANY_HELPERS. I still prefer
```
const stream = await navigator.mediaDevices.getUserMedia({audio: true})
const [ŧrack] = stream.getTracks();
```
Lines of code are cheap.

> Therefor we should design our tests not with that in mind.

We're much better of when taking into account the imperfection of implementations. In this case Firefox not liking addTrack without a stream. This shouldn't stop them from implementing that ofc but the compat hit should be in the addTrack() tests.